### PR TITLE
feat: updated empty state screens

### PR DESCRIPTION
**Pull request by BenjaminRuby**
## Context

Jira Issue: 

https://tools.boomerangplatform.net/jira/console/browse/ISEEC-610

### CODE REVIEW

File: src/Features/Teams/Teams.module.scss [hunk 1]
```diff
@@ -3,3 +3,7 @@
   right: 2rem;
   top: 12rem;
 }
+
+.tableRow:hover {
```
Comment (timrbula) [inline src/Features/Teams/Teams.module.scss:7]: unrelated fix for the team table that is clickable

File: src/Features/Workflows/WorkflowCard/UpdateWorkflow/UpdateWorkflow.tsx [hunk 2]
```diff
@@ -38,15 +38,15 @@ const UpdateWorkflow: React.FC<UpdateWorkflowProps> = ({ teamId, workflowId, onC
         containerClassName: styles.container,
       }}
       modalHeaderProps={{
-        title: "Update .json file",
+        title: "Update Workflow",
```
Comment (timrbula) [inline src/Features/Workflows/WorkflowCard/UpdateWorkflow/UpdateWorkflow.tsx:41]: Unrelated updates to this copy that I came across when testing and wanted to update to make more sense after receiving a few complaints.

File: src/Components/EmptyState/EmptyState.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import { ErrorPage } from "@boomerang-io/carbon-addons-boomerang-react";
+import EmptyGraphic from "./EmptyGraphic";
+
+const EmptyState: React.FC<any> = (props) => {
+  return <ErrorPage graphic={<EmptyGraphic />} {...props} />;
```
Comment (timrbula) [inline src/Components/EmptyState/EmptyState.tsx:6]: If we are mainly setting `header` to null and have the same empty search message in multiple places do we want to set those as the default props?

File: src/Features/GlobalProperties/PropertiesTable/PropertiesTable.tsx [hunk 4]
```diff
@@ -213,32 +213,7 @@ function PropertiesTable({ properties }: { properties: Property[] }) {
...
-                        ))}
-                      </TableRow>
-                    </TableHead>
-                  </Table>
-                </TableContainer>
-              )}
-            />
-            <Error404 header={null} title="No properties to be found" message={null} />
-          </>
+          <EmptyState title="Looks like there's nothing here" message="No results from that search, try again." />
```
Comment (timrbula) [inline src/Features/GlobalProperties/PropertiesTable/PropertiesTable.tsx:216]: Related to the previous comment. This looks like it works fine without the `header` being explicitly set to null. Makes sense bc I don't there is a default for the ErrorPage like there is the 404.

File: src/Features/Quotas/TeamQuotasOverview/TeamQuotasOverview.tsx [hunk 5]
```diff
@@ -36,7 +30,9 @@ export function TeamQuotasOverview({ teams }) {
   const teamData = teams.find((team) => team.id === teamId);
 
   if (error)
-    return <Error404 header="Team not found" title="Crikey. We can't find the team you are looking for." message="" />;
+    return (
+      <EmptyState header="Team not found" title="Crikey. We can't find the team you are looking for." message="" />
```
Comment (timrbula) [inline src/Features/Quotas/TeamQuotasOverview/TeamQuotasOverview.tsx:34]: Maybe we should make this consistent with the other error messages and use a title and a message instead of header and message?

File: src/Features/TaskTemplates/TaskTemplateOverview/TaskTemplateOverview.tsx [hunk 6]
```diff
@@ -354,7 +354,7 @@ export function TaskTemplateOverview({ taskTemplates, updateTemplateInState }) {
 
   if (templateNotFound)
     return (
-      <Error404
+      <EmptyState
         header="Task Template not found"
```
Comment (timrbula) [inline src/Features/TaskTemplates/TaskTemplateOverview/TaskTemplateOverview.tsx:358]: Same comment as above - maybe make this consistent by only using title and message?

File: src/Features/TeamDetailed/Settings/Settings.tsx [hunk 7]
```diff
@@ -30,8 +31,12 @@ export default function Settings({ team }: { team: FlowTeam }) {
     queryFn: resolver.query(teamsUrl),
   });
 
-  if (error) {
-    return <Error404 />;
+  if (!error) {
```
Comment (timrbula) [inline src/Features/TeamDetailed/Settings/Settings.tsx:34]: Updated this to show the the error message

File: src/Features/Editor/ChangeLog/ChangeLogTable/ChangeLogTable.tsx [hunk 8]
```diff
@@ -146,7 +146,7 @@ class ChangeLogTable extends Component<ChangeLogTableProps> {
             />
           </>
         ) : (
-          <Error404 header={null} message={null} title="No change logs found" />
+          <EmptyState header={null} message={null} title="No change logs found" />
```
Comment (timrbula) [inline src/Features/Editor/ChangeLog/ChangeLogTable/ChangeLogTable.tsx:149]: Depending on what you do for the defaults, one or two of these props aren't needed right?

File: src/Features/TeamProperties/TeamPropertiesTable/TeamPropertiesTable.tsx [hunk 9]
```diff
@@ -238,30 +239,8 @@ const TeamPropertiesTable: React.FC<TeamPropertiesTableProps> = ({
           </>
         ) : (
           <>
```
Comment (timrbula) [inline src/Features/TeamProperties/TeamPropertiesTable/TeamPropertiesTable.tsx:241]: Removed showing the empty table headers

File: src/Features/Workflows/WorkflowCard/WorkflowCard.tsx [hunk 10]
```diff
@@ -107,22 +107,21 @@ const WorkflowCard: React.FC<WorkflowCardProps> = ({ teamId, quotas, workflow })
 
   const menuOptions = [
     {
-      itemText: "Edit Workflow",
+      itemText: "Edit",
```
Comment (timrbula) [inline src/Features/Workflows/WorkflowCard/WorkflowCard.tsx:110]: Rearranged these and updated the copy per a request. I think this makes more sense

**Review by timrbula — CHANGES_REQUESTED** [review]
I made a few small changes that I commented on. Left some questions about default props and unnecessary props.

File: src/Features/Workflows/WorkflowCard/UpdateWorkflow/UpdateWorkflow.tsx (hunk 2 above)
Comment (BenjaminRuby) [reply src/Features/Workflows/WorkflowCard/UpdateWorkflow/UpdateWorkflow.tsx:41]: I agree that it could be confusing if you are not familiar with the json structure of a workflow and the import/export process. I feel like having `edit workflow` and `update workflow` menu options could also be a little confusing. Perhaps `Overwrite workflow`? Or something to help make it distinct that this is how you can import json and create a new revision?

File: src/Features/GlobalProperties/PropertiesTable/PropertiesTable.tsx [hunk 11]
```diff
@@ -237,7 +237,7 @@ function PropertiesTable({ properties }: { properties: Property[] }) {
                 </TableContainer>
               )}
             />
-            <Error404 header={null} title="No properties to be found" message={null} />
+            <EmptyState title="Looks like there's nothing here" message="No results from that search, try again." />
```
[the remark this answers is not in the document]
Comment (BenjaminRuby) [reply src/Features/GlobalProperties/PropertiesTable/PropertiesTable.tsx]: I'll clean these up a bit. EmptyState is just a wrapper around our ErrorPage component (which does not have any default props defined - it should not show a header if one is not specified)

**Review by timrbula — APPROVED** [review]
Looking good to me
